datadog/tests/resource_datadog_webhook_custom_variable_test.go

```diff
@@ -0,0 +1,115 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccDatadogWebhookCustomVariable_Basic(t *testing.T) {
+	ctx, accProviders := testAccProviders(context.Background(), t)
+	uniqueName := strings.ToUpper(strings.ReplaceAll(uniqueEntityName(ctx, t), "-", "_"))
+	accProvider := testAccProvider(t, accProviders)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: accProviders,
+		CheckDestroy:      testAccCheckDatadogWebhookCustomVariableDestroy(accProvider),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDatadogWebhookCustomVariableBasicConfig(uniqueName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogWebhookCustomVariableExists(accProvider, "datadog_webhook_custom_variable.foo"),
+					resource.TestCheckResourceAttr("datadog_webhook_custom_variable.foo", "name", uniqueName),
+					resource.TestCheckResourceAttr("datadog_webhook_custom_variable.foo", "value", "test-value"),
+					resource.TestCheckResourceAttr("datadog_webhook_custom_variable.foo", "is_secret", "false"),
+				),
+			},
+			{
+				Config: testAccCheckDatadogWebhookCustomVariableBasicConfigUpdated(uniqueName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogWebhookCustomVariableExists(accProvider, "datadog_webhook_custom_variable.foo"),
+					resource.TestCheckResourceAttr("datadog_webhook_custom_variable.foo", "name", uniqueName+"UPDATED"),
+					resource.TestCheckResourceAttr("datadog_webhook_custom_variable.foo", "value", "test-value-updated"),
```

Comment:
Hmm you shouldn't be able to set the updated value when it's a secret since it's not returned by the API. This should be failing no ?

Reply:
Thanks as discussed outside of Github issue, the field was marked as sensitive.